api/src/main/java/org/pmiops/workbench/firecloud/FireCloudServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package org.pmiops.workbench.firecloud;
+
+import javax.servlet.http.HttpServletResponse;
+import org.pmiops.workbench.firecloud.api.ProfileApi;
+import org.pmiops.workbench.firecloud.model.Me;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Service
+public class FireCloudServiceImpl implements FireCloudService {
+
+  @Override
+  public boolean isRequesterEnabledInFirecloud() throws ApiException {
+    Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+    ProfileApi profileApi = new ProfileApi();
+    ApiClient apiClient = new ApiClient();
+    apiClient.setAccessToken((String) authentication.getCredentials());
+    profileApi.setApiClient(apiClient);
+
+    Me me = profileApi.me();
+    // Users can only use FireCloud if all three flags are enabled.
+    return me.getEnabled() != null && isTrue(me.getEnabled().getAllUsersGroup())
+        && isTrue(me.getEnabled().getGoogle()) && isTrue(me.getEnabled().getLdap());
+  }
+
+  private boolean isTrue(Boolean b) {
```

Comment:
I'm surprised this is necessary, but maybe I've been in untyped Python too long? (Isn't `null` falsey?) Would this work? `me.getEnabled() && me.getEnabled().getAllUsersGroup() && me.getEnabled().getGoogle() && me.getEnabled().getLdap()`